ion-schema/src/isl/isl_constraint.rs

```diff
@@ -730,6 +741,29 @@ impl IslConstraintImpl {
                     }
                 }
             }
+            "ieee754_float" => {
+                if value.is_null() {
+                    return invalid_schema_error(
+                        "expected a symbol for an `ieee754_float` constraint, found null",
+                    );
+                }
+                if !value.annotations().is_empty() {
+                    return invalid_schema_error(
+                        "`ieee_754_float` constraint must be one of binary16, binary32, binary64",
+                    );
+                }
+                let string_value =
+                    value
+                        .as_symbol()
+                        .map(|s| s.text().unwrap())
+                        .ok_or_else(|| {
+                            invalid_schema_error_raw(format!(
+                                "expected ieee754_float to contain a symbol but found: {}",
+                                value.ion_type()
```

Comment:
suggestion—more more actionable error message, and also applicable for `null.symbol`.
```suggestion
                                "expected ieee754_float to be one of 'binary16', 'binary32', or 'binary64', but it was: {value}",
```